### ISSUE TITLE
Losing spaces before cross reference to table

### ISSUE THREAD

**Issue by arieknarie** [elapouya/python-docx-template#227]
In some cases a single or even double space is lost before a cross reference to a table. In the template document it is there, in the exported document it is gone.

**Comment by sebbelese** [comment]
I also experience whitespaces lost. Even in the middle of text which is originally in the template, when located "close" to a tag. I cannot identify to what it is related. Non-breaking spaces do not disappear.

**Comment by ram16230** [comment]
I also have the same problem. Whitespaces are missing before an opening tag and after an ending tag.

**Comment by elapouya** [comment]
Please provide a minimal test case (template+python code)

**Comment by arieknarie** [comment]
Thank you for looking into this. I will make a minimal case asap.

**Comment by arieknarie** [comment]
That took some trial and error, but please find below the python code _losing_spaces_code.text_ and the template _template_losing_spaces.docx_ . 

After replacement of the tags the space before the word 'spicy' has disappeared. The result depends really on the actual words, not on the length (in characters) of the strings. I hope you can find a solution to further improve your outstanding code!


[losing_spaces_code.txt](https://github.com/elapouya/python-docx-template/files/4605632/losing_spaces_code.txt)



[template_losing_spaces.docx](https://github.com/elapouya/python-docx-template/files/4605628/template_losing_spaces.docx)

**Comment by elapouya** [comment]
I reproduced the problem, but I cannot figure out what's going on : 
docxtpl did the job as expected : when looking at generated XML code,
`{{tag_1}}` and `{{tag_2}}` have been properly changed. And that's the ONLY thing that was changed into the XML : the space is still there between 'for' and 'spicy'. To summary, if spaces exist in the template, it should exist in output too !
The only thing I noticed in the template XML code is that between the 'for' and the 'spicy' words there are some manipulation : your text is not in the same run : may be because some style/color changes has occured (which are supported by docxtpl). 
If I create a template myself like yours, no problem occured because it is on the same run (same style).
Now, if I take your template, I unzip, edit document.xml, replace myself with a text editor `{{tag_1}}` and `{{tag_2}}`, then re-zip to create the modified docx : I have the same problem : a space has disappear ed.
It looks like a MS Word "issue", nothing I can fix as the output generated with docxtpl is relevant.
I have no idea why MS Word (and libre office too) strips a space on the screen despite it is still here in the XML code !
If someone has got an idea...

**Comment by elapouya** [comment]
As a workaround, I open the template on Libre Office (Star Office) and save it at once with another name : There is no more 'space' problem. When looking into de XML code, I noticed that libre office cleared all the suspicious XML tags that MS Word added between 'for' and 'spicy'

**Comment by arieknarie** [comment]
Thanks for all your help! I will try that  on the original document.

**Comment by spanishpy** [comment]
I was facing the same issue and after digging into the code I found the problem. Long story short, the underlying XML of the word document represents the text using multiple `<w:t>` tags. Text tags within a `{{ }}` or `{% %}` block are then removed and their attributes get lost. I've opened a [pull request](https://github.com/elapouya/python-docx-template/pull/278).
In the picture below, the left pane is the underlying XML. Middle pane is the resulting XML without the code changes. Right pane is the resulting XML with the code changes. Note the `xml:space="preserve"` is there now.
![Screenshot 2020-05-23 at 09 57 26](https://user-images.githubusercontent.com/9876012/82726577-f972a600-9cdc-11ea-8cea-ca407f561c5e.png)
